build-gcc.sh

```diff
@@ -0,0 +1,58 @@
+#!/bin/bash
+
+set -e
+
+GCC_VERSION=10.4.0
+GCC_SHA256=ab1974017834430de27fd803ade4389602a7d6ca1362496c57bef384b2a4cb07
+PREFIX="${PREFIX:-/usr}"
+
+prepare_sources() {
+    local archive=$1
+    local extension="${archive##*.}"
+    local tar_arg
+
+    case $extension in
+      xz)
+        tar_arg=J
+        ;;
+      bz2)
+        tar_arg=j
+        ;;
+      gz)
+        tar_arg=z
+        ;;
+      *)
+        echo "Unsupported archive format $extension"
+        exit 1
+        ;;
+    esac
+
+    tar x${tar_arg}f ${archive}
+}
+
+compile_with_autoconf() {
+    [ -e /etc/redhat-release ] && configure_args="--libdir=$PREFIX/lib64" || true
```

Comment:
I think there should be additional check here to ensure the architecture is 64bit. We also have `rpm-armhf` (32bit ARM) builder where this shouldn't apply.

Is there a specific reason why this feature is not added to the `rpm-armhf` builder?

Reply:
I've added a check to set up `--libdir=$PREFIX/lib64` or `--libdir=$PREFIX/lib` according to the architecture.

We have not added GCC in the `rpm-armhf` builder yet because we experienced a build issue:

```
checking for aarch64-unknown-linux-gnu-gcc... /gcc-10.4.0/host-aarch64-unknown-linux-gnu/gcc/xgcc -B/gcc-10.4.0/host-aarch64-unknown-linux-gnu/gcc/ -B/opt/gcc-10.4.0/aarch64-unknown-linux-gnu/bin/ -B/opt/gcc-10.4.0/aarch64-unknown-linux-gnu/lib/ -isystem /opt/gcc-10.4.0/aarch64-unknown-linux-gnu/include -isystem /opt/gcc-10.4.0/aarch64-unknown-linux-gnu/sys-include   -fno-checking
checking for suffix of object files... configure: error: in `/gcc-10.4.0/aarch64-unknown-linux-gnu/libgcc':
configure: error: cannot compute suffix of object files: cannot compile